pkg/service/service.go

```diff
@@ -333,3 +345,169 @@ func NewIndexingService(blobIndexLookup BlobIndexLookup, claimLookup ClaimLookup
 	}
 	return is
 }
+
+func PublishClaim(ctx context.Context, blobIndex BlobIndexLookup, claimLookup ClaimLookup, provIndex ProviderIndex, provider peer.AddrInfo, claim delegation.Delegation) error {
+	return publishIndexClaim(ctx, blobIndex, claimLookup, provIndex, provider, claim)
+}
+
+func publishIndexClaim(ctx context.Context, blobIndex BlobIndexLookup, claimLookup ClaimLookup, provIndex ProviderIndex, provider peer.AddrInfo, claim delegation.Delegation) error {
+	caps := claim.Capabilities()
+	if len(caps) == 0 {
+		return fmt.Errorf("missing capabilities in claim: %s", claim.Link())
+	}
+
+	if caps[0].Can() != assert.Index.Can() {
+		return fmt.Errorf("unsupported claim: %s", caps[0].Can())
+	}
+
+	reader := schema.Struct[assert.IndexCaveats](datamodel.IndexCaveatsType(), nil)
+	nb, rerr := reader.Read(caps[0].Nb())
+	if rerr != nil {
+		return fmt.Errorf("reading index claim data: %w", rerr)
+	}
+
+	results, err := provIndex.Find(ctx, providerindex.QueryKey{
+		Hash:         asCID(nb.Index).Hash(),
+		TargetClaims: []multicodec.Code{metadata.LocationCommitmentID},
+	})
+	if err != nil {
+		return fmt.Errorf("finding location commitment: %w", err)
+	}
+	if len(results) == 0 {
+		return fmt.Errorf("no location commitments found for index: %s", nb.Index)
+	}
+
+	var idx blobindex.ShardedDagIndex
+	var ferr error
+	for _, r := range results {
+		idx, ferr = fetchBlobIndex(ctx, blobIndex, claimLookup, r)
+		if ferr != nil {
+			continue
+		}
+		break
+	}
+	if ferr != nil {
+		return fmt.Errorf("fetching blob index: %w", err)
```

Comment:
it seems like if we had a successful fetch,  even if another errored, then we can proceed, no?

Reply:
Oops! Good catch.

Reply:
No wait, a successul fetch will set `ferr` to `nil`, so we will proceed.